### ISSUE TITLE
install on python 3.9  failed

### ISSUE THREAD

**Issue by hitbuyi** [enthought/mayavi#1176]
I have tried mayavi4.7.3+VTK9.0.2 in python3.9.5 on ubuntu20.04, it doesn't work
What's best version of VTK and mayaiv for python3.9.5(ubuntu20.04)?
```

(py1121) hitbuyi@hitbuyi-Dell-G15-5511:~/MachineLearning/3D/vis_3D/mayavi$ python setup.py install 
/home/hitbuyi/MachineLearning/3D/vis_3D/mayavi/setup.py:13: DeprecationWarning: 

  `numpy.distutils` is deprecated since NumPy 1.23.0, as a result
  of the deprecation of `distutils` itself. It will be removed for
  Python >= 3.12. For older Python versions it will remain present.
  It is recommended to use `setuptools < 60.0` for those Python versions.
  For more details, see:
    https://numpy.org/devdocs/reference/distutils_status_migration.html 


  from numpy.distutils.command import build, install_data
running install
/home/hitbuyi/.conda/envs/py1121/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
running build
----------------------------------------------------------------------
Building TVTK classes... vtkContextDevice2D: Ignoring method: Get/SetViewportRect
default: [293719408, 32704, 293719408, 32704], range: None
vtkContextDevice2D: Ignoring method: Get/SetViewportSize
default: [293719408, 32704], range: None
vtkEventDataForDevice: Ignoring method: Get/SetAction
default: -1, range: None
vtkEventDataForDevice: Ignoring method: Get/SetDevice
default: -1, range: None
vtkFXAAOptions: Ignoring method: Get/SetDebugOptionValue
default: 0, range: None
vtkIncrementalForceLayout: Ignoring method: Get/SetGravityPoint
default: [200.0, 200.0], range: None
vtkOpenGLBufferObject: Ignoring method: Get/SetType
default: 0, range: None
vtkOpenGLFXAAFilter: Ignoring method: Get/SetDebugOptionValue
default: 0, range: None
vtkRenderWidget: Ignoring method: Get/SetPosition
default: [0, 0], range: None
vtkRenderWidget: Ignoring method: Get/SetSize
default: [300, 300], range: None
vtkShader: Ignoring method: Get/SetType
default: 3, range: None
vtkVolumeTexture: Ignoring method: Get/SetPartitions
default: [1, 1, 1], range: None
vtkChartMatrix: Ignoring method: Get/SetGutter
default: [15.0, 15.0], range: None
vtkChartMatrix: Ignoring method: Get/SetSize
default: [0, 0], range: None
vtkContextArea: Ignoring method: Get/SetDrawAreaBounds
default: [0.0, 0.0, 300.0, 300.0], range: None
vtkContextArea: Ignoring method: Get/SetDrawAreaResizeBehavior
default: 0, range: None
vtkContextArea: Ignoring method: Get/SetFixedMargins
default: [0, 0, 0, 0], range: None
vtkContextArea: Ignoring method: Get/SetFixedRect
default: [0, 0, 300, 300], range: None
vtkContextArea: Ignoring method: Get/SetGeometry
default: [0, 0, 300, 300], range: None
vtkImplicitProjectOnPlaneDistance: Ignoring method: Get/SetNorm
default: 2, range: None
vtkOpenGLVertexBufferObject: Ignoring method: Get/SetCoordShiftAndScaleMethod
default: 0, range: None
vtkSimple3DCirclesStrategy: Ignoring method: Get/SetMarkedValue
default: (invalid), range: None
vtkArrayNorm: Ignoring method: Get/SetWindow
default: [0, 9223372036854775807), range: None
vtkChart: Ignoring method: Get/SetSize
default: [0.0, 0.0, 0.0, 0.0], range: None
vtkChartXYZ: Ignoring method: Get/SetAxisColor
default: [0, 0, 0, 255], range: None
Warning: We cannot determine the trait type of vtkHigherOrderHexahedron.Order
Warning: We cannot determine the trait type of vtkHigherOrderQuadrilateral.Order
Segmentation fault (core dumped)
```

**Comment by hitbuyi** [comment]
I have tried to installed  setuptools of version 59.1.0, it doesn't solve the problem

**Comment by larsoner** [comment]
Do you see this failure on https://github.com/enthought/mayavi/pull/1180 ? If so, try installing mayavi on that branch with the env var `VTK_PARSER_VERBOSE=1`, it should tell you what is causing the segfault

**Comment by prabhuramachandran** [comment]
This is now fixed with the new 4.8.1 release.
